### PR TITLE
configure controllers tui does not support when installing desktop

### PR DESCRIPTION
**Pull request by mwhudson**
We could definitely make this more elegant. But what do you think?

### CODE REVIEW

File: subiquity/ui/views/source.py [hunk 1]
```diff
@@ -54,6 +55,7 @@ def __init__(self, controller, sources, current_id, search_drivers: bool):
                 ns[source.id] = RadioButtonField(
                     group, source.name, '\n' + source.description)
                 initial[source.id] = source.id == current_id
+                self.sources_by_id[source.id] = source
```
Comment (dbungert) [inline subiquity/ui/views/source.py:58]: A unit test on population of sources_by_id would be nice to have.

File: subiquity/ui/views/source.py [hunk 2]
```diff
@@ -93,7 +95,8 @@ def done(self, result):
             if k == "search_drivers":
                 continue
             if v:
-                self.controller.done(k, search_drivers=search_drivers)
+                self.controller.done(
```
Comment (dbungert) [inline subiquity/ui/views/source.py:98]: This done method is a bit unclear but that's not new to this PR.

**Review by dbungert — APPROVED** [review]

**Comment by mwhudson** [comment]
I accidentally rewrote this to be more generic, more local and probably even dirtier...

**Comment by mwhudson** [comment]
I changed this again in a way that probably depends on #1483 but I'm thoroughly confused by now.

**Comment by mwhudson** [comment]
OK back to a variant (ha!) that doesn't depend on understanding the difference between the two variants.

**Review by dbungert — APPROVED** [review]

**Pull request merged**